### PR TITLE
Bombard Peter with a thousand different changes in a single PR

### PR DESCRIPTION
**Pull request by Grobobobo**
At least the commits are named properly this time

### CODE REVIEW

File: LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2LWAbilitiesModTemplate.uc [hunk 1]
```diff
@@ -533,6 +539,25 @@ static function UpdateRevive(X2AbilityTemplate Template)
 	ExcludeEffects.AddExcludeEffect(class'X2StatusEffects'.default.BurningName, 'AA_UnitIsBurning');
 	Template.AbilityShooterConditions.AddItem(ExcludeEffects);
 }
+static function AddImmuneConditionToPoisonSpit(X2AbilityTemplate Template)
+{
+	local X2Condition_UnitImmunities				UnitImmunityCondition;
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2LWAbilitiesModTemplate.uc]: I don't think you need quite such a large amount of whitespace here 😆

File: LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2LWAbilitiesModTemplate.uc [hunk 2]
```diff
@@ -91,6 +91,12 @@ static function UpdateAbilities(X2AbilityTemplate Template, int Difficulty)
 		case 'BlastShield':
 			ReplaceWithDamageReductionExplosive(Template);
 			break;
+		case 'PoisonSpit':
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2LWAbilitiesModTemplate.uc]: Could you please merge this commit with the previous one?

File: LongWarOfTheChosen/Src/LW_Overhaul/Classes/SniperDefense_UIScreenListener.uc [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+			SourceUnitState = XComGameState_Unit(History.GetGameStateForObjectID(ActivatedAbilityStateContext.InputContext.SourceObject.ObjectID));
+			WeaponState = XComGameState_Item(GameState.GetGameStateForObjectID(ActivatedAbilityStateContext.InputContext.ItemObject.ObjectID));
+			TargetedUnitState = XComGameState_Unit(History.GetGameStateForObjectID(ActivatedAbilityStateContext.InputContext.PrimaryTarget.ObjectID));
+            SoundRange = WeaponState.GetItemSoundRange();
+
+            if( SoundRange > 0 && TargetedUnitState.ObjectID > 0 && WeaponState.SoundOriginatesFromOwnerLocation() && !SourceUnitState.ControllingPlayerIsAI())//There is a valid target and the Unit firing is player controlled
+			{			
+				`Log("Sniper Defense Weapon sound generated by XCom");
+				nEnemiesVisible = TargetedUnitState.GetNumVisibleEnemyUnits(true, true,,,,, false);
+				If(nEnemiesVisible == 0) //No visible enemies than this is a concealed spotter/sniper shot
```
[the remark this answers is not in the document]
Comment (Grobobobo) [reply LongWarOfTheChosen/Src/LW_Overhaul/Classes/SniperDefense_UIScreenListener.uc]: Do you mean the nEnemiesVisible variable, or the unnescessary "Sniper Defense Weapon sound generated by XCom"?
Not exactly sure what you mean here

**Comment by pledbrook** [comment]
Merged manually after a rebase

**Pull request closed without merging**